array.go

```diff
@@ -573,70 +573,78 @@ func (a *arrayObject) estimateMemUsage(ctx *MemUsageContext) (uint64, error) {
 			continue
 		}
 
-		inc, err := a.values[i].MemUsage(ctx)
+		inc, newInc, err := a.values[i].MemUsage(ctx)
 		samplesVisited += 1
-		total += inc
+		memUsage += inc
+		newMemUsage += newInc
 		// average * number of a.values
-		runningEstimate = uint64((float32(total) / float32(samplesVisited)) * float32(len(a.values)))
+		runningEstimate = uint64((float32(memUsage) / float32(samplesVisited)) * float32(len(a.values)))
+		newRunningEstimate = uint64((float32(newMemUsage) / float32(samplesVisited)) * float32(len(a.values)))
 		if err != nil {
-			return runningEstimate, err
+			return runningEstimate, newRunningEstimate, err
 		}
-		if exceeded := ctx.MemUsageLimitExceeded(total); exceeded {
-			return total, nil
+		if exceeded := ctx.MemUsageLimitExceeded(memUsage); exceeded {
```

Comment:
[q] should we use `mewMemUsage` instead of `memUsage`?

Reply:
Good question, eventually the newMem and mem will have the same value but we are using this PR to collect data about the gap between the two values for customers to get an idea of the impact of this change. Until we apply this new changes all the memory logic will rely on the current memUsage.